### PR TITLE
better primitives docs and comments, and some tweaks

### PR DESCRIPTION
**Pull request by rafaqz**
@skygering let me know if this makes more sense to you.

Explaining recursive methods is kind of hard, and I write a lot of code like this so I probably don't see how or what is confusing.

### CODE REVIEW

File: src/primitives.jl [hunk 1]
```diff
@@ -12,14 +12,67 @@ $CALC_EXTENT_KEYWORD
 
 # This file mainly defines the [`apply`](@ref) function.
 
+#=
+## What is `apply`?
+
+`apply` applies some function to every geometry matching the `Target`
+GeoInterface trait, in some arbitrarily nested object made up of:
+- `AbstractArray`s 
+- Some arbitrary iterables may also work here
```
Comment (skygering) [inline src/primitives.jl:21]: I am not really sure what this is referring too

**Comment by skygering** [comment]
I did a spell check, but yeah other than my one comment above, I get what it does. Reads pretty well

File: src/primitives.jl (hunk 1 above)
Comment (rafaqz) [reply src/primitives.jl:21]: You mean the dot points? They are the types that will be accepted and recursed over by `apply`.

Maybe I can make that clearer.

Comment (rafaqz) [reply src/primitives.jl:21]: Or you mean the iterables? it doesn't quite sit with the others but anything unknown (not trait and not an AbstractArray) we try to iterate over.

Comment (skygering) [reply src/primitives.jl:21]: Okay yeah that makes sense.

**Comment by skygering** [comment]
Happy for it to merge

File: src/primitives.jl (hunk 1 above)
Comment (rafaqz) [reply src/primitives.jl:21]: Ok I'll try to work that into the text

Comment (rafaqz) [inline src/primitives.jl:21]:
```suggestion
- `AbstractArray`s (we also try to iterate other non-GeoInteface compatible object)
```

File: src/primitives.jl [hunk 2]
```diff
@@ -12,14 +12,66 @@ $CALC_EXTENT_KEYWORD
...
+## What is `apply`?
+
+`apply` applies some function to every geometry matching the `Target`
+GeoInterface trait, in some arbitrarily nested object made up of:
+- `AbstractArray`s (we also try to iterate other non-GeoInteface compatible object)
+- `FeatureCollectionTrait` objects
+- `FeatureTrait` objects
+- `AbstractGeometryTrait` objects
+
+It recursively calls `apply` through these nested
```
Comment (rafaqz) [inline src/primitives.jl:25]:
```suggestion
`apply` recursively calls itself through these nested
```

File: src/primitives.jl [hunk 3]
```diff
@@ -12,14 +12,66 @@ $CALC_EXTENT_KEYWORD
...
+
+`apply` applies some function to every geometry matching the `Target`
+GeoInterface trait, in some arbitrarily nested object made up of:
+- `AbstractArray`s (we also try to iterate other non-GeoInteface compatible object)
+- `FeatureCollectionTrait` objects
+- `FeatureTrait` objects
+- `AbstractGeometryTrait` objects
+
+`apply` recursively calls itself through these nested
+layers until it reaches the `Target`, where it applies `f`, and stops.
```
Comment (rafaqz) [inline src/primitives.jl:26]:
```suggestion
layers until it reaches objects with the `Target` GeoInterface trait. When found `apply` applies the function `f`, and stops.
```

**Pull request merged**